lib/utilities/MD5Sum.js

```diff
@@ -0,0 +1,24 @@
+const Transform = require('stream').Transform;
+const crypto = require('crypto');
+
+class MD5Sum extends Transform {
+
+    constructor(callback) {
+        super({});
+        this.hash = crypto.createHash('md5');
+        this.callback = callback;
+    }
+
+    _transform(chunk, encoding, callback) {
```

Comment:
let's call this callback something else so as not to be confused with the callback in the constructor -- done or cb